### PR TITLE
Add a note about HR not being a replacement for ORM

### PR DESCRIPTION
**Pull request by gsmet**
As discussed in https://quarkusio.zulipchat.com/#narrow/stream/191168-core-team/topic/negative.20feedback

@Sanne @DavideD @geoand @cescoffier I let you give me feedback on the wording. I tried to be as neutral as possible, feedback very welcome.

Created as draft for now as I want everybody on board with this before merging.

### CODE REVIEW

File: docs/src/main/asciidoc/hibernate-reactive-panache.adoc [hunk 1]
```diff
@@ -14,6 +14,16 @@ breadth of an Object Relational Mapper allowing you to access your database over
 It makes complex mappings possible, but it does not make simple and common mappings trivial.
 Hibernate Reactive with Panache focuses on making your entities trivial and fun to write in Quarkus.
 
+[NOTE]
+====
+Hibernate Reactive is not a replacement for xref:hibernate-orm-panache.adoc[Hibernate ORM] or the future of Hibernate ORM.
+It is a different stack tailored for reactive use cases.
+
+Also, using RESTEasy Reactive, our default REST layer, does not mandate the use of Hibernate Reactive.
```
Comment (geoand) [inline docs/src/main/asciidoc/hibernate-reactive-panache.adoc:22]:
```suggestion
Furthermore, using RESTEasy Reactive, our default REST layer, does not mandate the use of Hibernate Reactive.
```

**Comment by geoand** [comment]
Very nice! 

I added a small suggestion (which can be applied to both notes)

File: docs/src/main/asciidoc/hibernate-reactive-panache.adoc [hunk 2]
```diff
@@ -14,6 +14,16 @@ breadth of an Object Relational Mapper allowing you to access your database over
 It makes complex mappings possible, but it does not make simple and common mappings trivial.
 Hibernate Reactive with Panache focuses on making your entities trivial and fun to write in Quarkus.
 
+[NOTE]
+====
+Hibernate Reactive is not a replacement for xref:hibernate-orm-panache.adoc[Hibernate ORM] or the future of Hibernate ORM.
+It is a different stack tailored for reactive use cases.
```
Comment (cescoffier) [inline docs/src/main/asciidoc/hibernate-reactive-panache.adoc:20]:
```suggestion
It is a different stack tailored for reactive use cases where you need high-concurrency.
```

File: docs/src/main/asciidoc/hibernate-reactive-panache.adoc [hunk 3]
```diff
@@ -14,6 +14,16 @@ breadth of an Object Relational Mapper allowing you to access your database over
...
+
+Also, using RESTEasy Reactive, our default REST layer, does not mandate the use of Hibernate Reactive.
+It is perfectly valid to use RESTEasy Reactive with Hibernate ORM,
+and if you are not accustomed with the reactive paradigms, it is in most cases recommended to use Hibernate ORM.
```
Comment (cescoffier) [inline docs/src/main/asciidoc/hibernate-reactive-panache.adoc:24]:
```suggestion
and if you do not need high-concurrency, or are not accustomed to the reactive paradigm, it is in most cases recommended to use Hibernate ORM.
```

**Review by cescoffier — DISMISSED** [review]

**Comment by gsmet** [comment]
I included all your suggestions.

**Review by geoand — APPROVED** [review]

File: docs/src/main/asciidoc/hibernate-reactive.adoc [hunk 4]
```diff
@@ -9,10 +9,19 @@ include::_attributes.adoc[]
...
-
 link:https://hibernate.org/reactive/[Hibernate Reactive] is a reactive API for Hibernate ORM, supporting non-blocking database drivers
 and a reactive style of interaction with the database.
 
+[NOTE]
+====
+Hibernate Reactive is not a replacement for xref:hibernate-orm.adoc[Hibernate ORM] or the future of Hibernate ORM.
+It is a different stack tailored for reactive use cases.
+
+Also, using RESTEasy Reactive, our default REST layer, does not mandate the use of Hibernate Reactive.
```
Comment (Sanne) [inline docs/src/main/asciidoc/hibernate-reactive.adoc:20]:
```suggestion
Also, using RESTEasy Reactive, our default REST layer, does not require the use of Hibernate Reactive.
```

**Review by Sanne — APPROVED** [review]
Cool :+1: 
Added a minor tone suggestion

File: docs/src/main/asciidoc/hibernate-reactive-panache.adoc [hunk 5]
```diff
@@ -14,6 +14,16 @@ breadth of an Object Relational Mapper allowing you to access your database over
...
 Hibernate Reactive with Panache focuses on making your entities trivial and fun to write in Quarkus.
 
+[NOTE]
+====
+Hibernate Reactive is not a replacement for xref:hibernate-orm-panache.adoc[Hibernate ORM] or the future of Hibernate ORM.
+It is a different stack tailored for reactive use cases where you need high-concurrency.
+
+Furthermore, using RESTEasy Reactive, our default REST layer, does **not** mandate the use of Hibernate Reactive.
+It is perfectly valid to use RESTEasy Reactive with Hibernate ORM,
+and if you do not need high-concurrency, or are not accustomed to the reactive paradigm, it is in most cases recommended to use Hibernate ORM.
```
Comment (cescoffier) [inline docs/src/main/asciidoc/hibernate-reactive-panache.adoc:24]: I would remove in the most cases. I think it's clear enough.

Comment (gsmet) [reply docs/src/main/asciidoc/hibernate-reactive-panache.adoc]: Done.

**Comment by gsmet** [comment]
I applied all your suggestions, this should be good to go.

**Pull request merged**